Added iOS specific section for OIDC Auth provider (#2613)
aws-amplify/docs: docs/lib/datastore/fragments/ios/setup-auth-rules/owner_based_auth_oidc.md

> When using a third-party OIDC auth provider, you are required to provide an instance of `APIAuthProviderFactory`. Your factory implementation must be capable of creating objects conforming to the `AmplifyOIDCAuthProvider` protocol.  The responsibility of the `AmplifyOIDCAuthProvider` is to return the JWT token that was provided by your OIDC provider. To do this:

Reviewer: Remove double whitespaces

> When using a 3rd party OIDC auth provider, you are required to provide an instance of `APIAuthProviderFactory` which is capable of creating an instance of a class that conforms to `AmplifyOIDCAuthProvider`.  The responsibility of the `AmplifyOIDCAuthProvider` is to return the JWT token that was provided by your OIDC provider.  To do this:
> * Create a subclass of `APIAuthProviderFactory`

Reviewer 2: ```suggestion
When using a third-party OIDC auth provider, you are required to provide an instance of `APIAuthProviderFactory`. Your factory implementation must be capable of creating objects conforming to the `AmplifyOIDCAuthProvider` protocol.  The responsibility of the `AmplifyOIDCAuthProvider` is to return the JWT token that was provided by your OIDC provider. To do this:
* Create a subclass of `APIAuthProviderFactory`
```

```swift
class MyAPIAuthProviderFactory: APIAuthProviderFactory {
    let myAuthProvider = MyOIDCAuthProvider()

    override func oidcAuthProvider() -> AmplifyOIDCAuthProvider? {
        return myAuthProvider
    }
}
```

> * Implement your class which conforms to `AmplifyOIDCAuthProvider`

Reviewer 2: ```suggestion
* Implement your class which conforms to `AmplifyOIDCAuthProvider`:
```

```swift
class MyOIDCAuthProvider : AmplifyOIDCAuthProvider {
    func getLatestAuthToken() -> Result<String, Error> {
       ....
    }
}
```

> * Finally, register your instance of `APIAuthProviderFactory` prior to calling `Amplify.configure()`

Reviewer 2: ```suggestion
* Finally, register your instance of `APIAuthProviderFactory` prior to calling `Amplify.configure()`:
```
